cardano-node/src/Cardano/Node/ToObjectOrphans.hs

```diff
@@ -0,0 +1,188 @@
+{-# LANGUAGE FlexibleContexts      #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE ScopedTypeVariables   #-}
+{-# LANGUAGE UndecidableInstances  #-}
+{-# OPTIONS_GHC -fno-warn-orphans  #-}
+
+module Cardano.Node.ToObjectOrphans
+  (
+    WithTip (..)
+  , showTip
+  , showWithTip
+  ) where
+
+import           Cardano.Prelude hiding (atomically, show)
+import           Prelude (String, show)
+
+import           Data.Aeson (Value (..), toJSON, (.=))
+import           Data.Text (Text, pack)
+import           Cardano.BM.Data.Tracer (ToObject (..),
+                                         TracingVerbosity (..),
+                                         Transformable (..),
+                                         emptyObject, mkObject, trStructured)
+
+import           Ouroboros.Consensus.Ledger.Abstract
+import           Ouroboros.Consensus.Util.Condense
+import qualified Ouroboros.Network.AnchoredFragment as AF
+import           Ouroboros.Network.Block
+import qualified Ouroboros.Storage.ChainDB as ChainDB
+import qualified Ouroboros.Storage.LedgerDB.OnDisk as LedgerDB
+
+
+-- | Tracing wrapper which includes current tip in the logs (thus it requires
+-- it from the context).
+--
+-- TODO: this should be moved to `ouroboros-consensus`.  Running in a seprate
+-- STM transaction we risk reporting  wrong tip.
+--
+data WithTip blk a =
+    WithTip
+      (Point blk)
+      -- ^ current tip point
+      a
+      -- ^ data
+
+showWithTip :: Condense (HeaderHash blk)
+            => (a -> String)
+            -> WithTip blk a
+            -> String
+showWithTip customShow (WithTip tip a) = "[" ++ (showTip tip) ++ "] " ++ customShow a
+
+showTip :: Condense (HeaderHash blk)
+        => Point blk
+        -> String
+showTip tip = case pointHash tip of
+    GenesisHash -> "genesis"
+    BlockHash h -> take 7 (condense h)
+
+instance ( Show a
+         , Condense (HeaderHash blk)
+         ) => Show (WithTip blk a) where
+
+    show = showWithTip show
+
+
+
+-- | instances of ToObject
+
+instance (Condense (HeaderHash blk), ProtocolLedgerView blk)
+            => Transformable Text IO (WithTip blk (ChainDB.TraceEvent blk)) where
+    trTransformer = trStructured -- structure required, will call 'toObject'
+instance (Condense (HeaderHash blk), ProtocolLedgerView blk)
+            => ToObject (WithTip blk (ChainDB.TraceEvent blk)) where
+    -- toObject MinimalVerbosity _ = emptyObject -- no output
```

Comment:
> Looking good, just some misc comments.
> 
> It would be great if you pasted the trace output for all three verbosity levels. Maybe we spot things that are too verbose for the given verbosity level, etc.

yes, let's do this in a second PR. This one should be integrated when ready, to lower the burden of rebasing with master (other changes coming in).

Reply:
this serves as an example of how to turn off tracing.

Reply:
> > Looking good, just some misc comments.
> > It would be great if you pasted the trace output for all three verbosity levels. Maybe we spot things that are too verbose for the given verbosity level, etc.
> 
> yes, let's do this in a second PR. This one should be integrated when ready, to lower the burden of rebasing with master (other changes coming in).

No, I mean: can you paste the output (for all three verbosities) that the node produces running *this PR*? (before we merge it) Otherwise, we're blindly merging something that might trace something totally  wrong.

I don't mind fine-tuning the trace output in follow-up PRs.

Reply:
> this serves as an example of how to turn off tracing.

Ah, in that case a comment mentioning this would be helpful, otherwise other people will also suspect that it is a stray comment.